src/Symfony/Component/Console/Attribute/Option.php

```diff
@@ -76,17 +76,13 @@ public static function tryFrom(\ReflectionParameter $parameter): ?self
         if ('bool' === $self->typeName) {
             $self->mode = InputOption::VALUE_NONE | InputOption::VALUE_NEGATABLE;
```

Comment:
I think negatable should only be added if ((nullable and default value is null) or (default value is true)).

Reply:
Kevin, could you explain a bit more about why those requirements are necessary? Normally boolean option will receive `false` directly if negable is used (see tests)

Reply:
A while back, I was confused by the purpose of negatable. I found it's only really practical when the default value is null. Consider the following example:

A bundle that has a `do:something` command. The bundle has a config to configure caching when this command is run - this is the default. For the command, we'd want --cache and --no-cache but if neither is passed, we want it to be null and use the bundle config in this case.

Also, adding negatable to every boolean option adds it to the command's help which, imo, can create confusion.

Reply:
I added support for nullable boolean options -> `#[Option] ?bool $foo = null`, so we can handle the 3 variants you described. However, supporting negatable non-nullable options is still a possibility, as we discussed on Slack.